src/flask/blueprints.py

```diff
@@ -344,17 +344,12 @@ def extend(bp_dict, parent_dict):
                 app.cli.add_command(self.cli)
 
         for blueprint, bp_options in self._blueprints:
-            bp_options = bp_options.copy()
```

Comment:
Is this required? I'm wasn't sure and the code added works as expected in Quart.

Reply:
Yes, otherwise you're mutating the shared state, so each parent registration will affect future registrations.